### PR TITLE
Add TCDM-Streamer combination test

### PR DESCRIPTION
**Pull request by rgantonio**
This PR adds a test that combines the TCDM and streamer. The test is as follows:

1. Load data through DMA ports first and double check if the data are correct
2. Configure streamer CSRs
3. Read data through the reader ports of the streamer. Also, verify the data at this step.
4. Write data using the streamer port.
5. Read the written data from step 4.

We add the following modifications:
- Template for the TCDM-streamer testbench
- Add wide data repacking function in `snax_util.py`
- Add TCDM-streamer test
- Add a wide data width parameter in the configuration file
- Re-factor `Makefile`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -71,8 +85,17 @@ gen_stream_tb:	$(STREAM_GEN_OUT_SCALA_FILE) $(STREAM_GEN_OUT_TOP_FILE) $(STREAM_
 	--tpl_path="${STREAM_GEN_TPL_TB_FILE}" \
 	--out_path="${STREAM_GEN_OUT_TB_FILE}"
 
+#-----------------------------
+# Generate Streamer-TCDM Wrapper Testbench
+#-----------------------------
+gen_stream_tcdm_tb:	$(STREAM_GEN_OUT_SCALA_FILE) $(STREAM_GEN_OUT_TOP_FILE) $(STREAM_GEN_OUT_RTL_FILE)
```
Comment (JosseVanDelm) [inline Makefile:91]: Should this recipe not have the name of the generated file?
```suggestion
$(STREAM_GEN_OUT_TB_FILE): $(STREAM_GEN_OUT_SCALA_FILE) $(STREAM_GEN_OUT_TOP_FILE) $(STREAM_GEN_OUT_RTL_FILE)
```

Comment (rgantonio) [reply Makefile:91]: Good point haha why did I not do that last time aksdjalksdjaslkdjaslkdjl what was I thinking 🍩

**Review by xiaoling-yi — APPROVED** [review]
Good job!

File: tests/cocotb/test_stream_tcdm.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+CSR_TEMPORAL_STRIDE_0 = 1
+CSR_TEMPORAL_STRIDE_1 = 2
+CSR_TEMPORAL_STRIDE_2 = 3
+CSR_SPATIAL_STRIDE_0 = 4
+CSR_SPATIAL_STRIDE_1 = 5
+CSR_SPATIAL_STRIDE_2 = 6
+CSR_BASE_PTR_0 = 7
+CSR_BASE_PTR_1 = 8
+CSR_BASE_PTR_2 = 9
+CSR_START_STREAMER = 10
```
Comment (xiaoling-yi) [inline tests/cocotb/test_stream_tcdm.py:60]: Does this duplicated a bit as in the `test_basic_streamer.py`?

Comment (rgantonio) [reply tests/cocotb/test_stream_tcdm.py:60]: It is I'm thinking of porting outside next time. Hmm maybe in the next PR 😄

**Pull request merged**